### PR TITLE
fix: Fix date/time related runtime code so fields show up correctly in UI

### LINKED ISSUE #205: Fix date/time related runtime code so fields show up correctly in UI
Part of epic #248 

See #200 

While field reader/writer can use individual field types like `FieldType.DATE_TIME_TZ`, it's annoying creating all the overload methods for all of those date/time related field types. Introduce `FieldType.ANY_DATE` which works as an umbrella field type so TypeConverter and FieldAction can consume/produce all those fields.

### PR DESCRIPTION
**Pull request by jpav**
Fixes #205

### CODE REVIEW

File: runtime/core/src/main/java/io/atlasmap/converters/DateConverter.java [hunk 1]
```diff
@@ -139,32 +135,104 @@ public Short toShort(Date value) throws AtlasConversionException {
...
     }
 
     @AtlasConversionInfo(sourceType = FieldType.DATE_TIME, targetType = FieldType.DATE_TIME)
-    public Timestamp toSqlTimestamp(Date date) throws AtlasConversionException {
+    public Timestamp toSqlTimestamp(Date date) {
         return Timestamp.from(date.toInstant());
     }
 
+    @AtlasConversionInfo(sourceType = FieldType.DATE_TIME_TZ, targetType = FieldType.DATE_TIME_TZ)
+    public ZonedDateTime toZonedDateTime(Calendar calendar) {
```
Comment (igarashitm) [inline runtime/core/src/main/java/io/atlasmap/converters/DateConverter.java]: Can you move this to [CalendarConverter](https://github.com/atlasmap/atlasmap/blob/master/runtime/core/src/main/java/io/atlasmap/converters/CalendarConverter.java)? I cleaned up those date/time converters to put same parameter ones together.

File: runtime/core/src/main/java/io/atlasmap/converters/DateConverter.java [hunk 2]
```diff
@@ -139,32 +135,104 @@ public Short toShort(Date value) throws AtlasConversionException {
...
+        return DateTimeHelper.toZonedDateTime(calendar);
+    }
+
+    @AtlasConversionInfo(sourceType = FieldType.DATE_TIME, targetType = FieldType.DATE_TIME_TZ)
+    public ZonedDateTime toZonedDateTime(Date date, String sourceFormat, String targetFormat) {
+        return DateTimeHelper.toZonedDateTime(date, sourceFormat == null ? targetFormat : sourceFormat);
+    }
+
+    @AtlasConversionInfo(sourceType = FieldType.DATE, targetType = FieldType.DATE_TIME_TZ)
+    public ZonedDateTime toZonedDateTime(LocalDate date, String sourceFormat, String targetFormat) {
```
Comment (igarashitm) [inline runtime/core/src/main/java/io/atlasmap/converters/DateConverter.java]: To the LocalDateConverter, ditto for others doesn't have `java.util.Date` as a parameter.

File: runtime/core/src/main/java/io/atlasmap/core/DefaultAtlasConversionService.java [hunk 3]
```diff
@@ -510,9 +506,7 @@ public FieldType fieldTypeFromClass(String className) {
         case DATE:
             return java.time.LocalDate.class;
         case DATE_TIME:
-            return java.util.Date.class;
-            // TODO do we prefer java.time.* at some point? - https://github.com/atlasmap/atlasmap/issues/312
-            // return java.time.LocalDateTime.class;
+            return java.time.LocalDateTime.class;
```
Comment (igarashitm) [inline runtime/core/src/main/java/io/atlasmap/core/DefaultAtlasConversionService.java]: Is this one required for any reason? I think java.util.Date is still a majority so I'd vote to stick with java.util.Date as a first representative for now.

File: runtime/core/src/test/java/io/atlasmap/core/DefaultAtlasConversionServiceTest.java [hunk 4]
```diff
@@ -678,24 +678,24 @@ public void testConvertTypeFromLong() throws AtlasConversionException {
...
-        assertEquals(BigDecimal.valueOf(1), service.convertType((Number)1, FieldType.NUMBER, FieldType.DECIMAL));
-        assertEquals(1.0d, service.convertType((Number)1, FieldType.NUMBER, FieldType.DOUBLE));
-        assertEquals(1.0f, service.convertType((Number)1, FieldType.NUMBER, FieldType.FLOAT));
-        assertEquals(1, service.convertType((Number)1, FieldType.NUMBER, FieldType.INTEGER));
-        assertEquals(1L, service.convertType((Number)1, FieldType.NUMBER, FieldType.LONG));
-        assertEquals(1, service.convertType((Number)1, FieldType.NUMBER, FieldType.NUMBER));
-        assertEquals((short)1, service.convertType((Number)1, FieldType.NUMBER, FieldType.SHORT));
-        assertEquals("1", service.convertType((Number)1, FieldType.NUMBER, FieldType.STRING));
-        assertEquals(LocalTime.class, service.convertType((Number)1, FieldType.NUMBER, FieldType.TIME).getClass());
+        assertEquals(BigInteger.valueOf(1), service.convertType(1, FieldType.NUMBER, FieldType.BIG_INTEGER));
```
Comment (igarashitm) [inline runtime/core/src/test/java/io/atlasmap/core/DefaultAtlasConversionServiceTest.java]: Any reason removing casting to Number? IIRC without this cast the NumberConverter is not picked up, instead IntegerConverter is used.

Comment (jpav) [reply runtime/core/src/test/java/io/atlasmap/core/DefaultAtlasConversionServiceTest.java]: Sorry, that was done automatically by a Save As action in my IDE. I've changed it a bit so the IDE doesn't mess it up.

**Pull request merged**